crawl-ref/source/directn.h

```diff
@@ -106,6 +106,15 @@ struct direction_chooser_args
 
 };
 
+struct dist_sorter
+{
+    coord_def pos;
+    bool operator()(const actor* a, const actor* b)
+    {
+        return a->pos().distance_from(pos) > b->pos().distance_from(pos);
+    }
+};
+
```

Comment:
directn.* is primarily for the direction_chooser and associated components, this struct belongs in spl-util.h since it is (presently) only used by spell-like things.